src/prpy/planning/base.py

```diff
@@ -52,13 +52,30 @@ def __init__(self, func):
 
     def __call__(self, instance, robot, *args, **kw_args):
         env = robot.GetEnv()
+        defer = kw_args.get('defer')
 
-        with Clone(env, clone_env=instance.env):
-            planning_traj = self.func(instance, Cloned(robot), *args, **kw_args)
-            traj = openravepy.RaveCreateTrajectory(env, planning_traj.GetXMLId())
-            traj.Clone(planning_traj, 0)
+        with Clone(env, clone_env=instance.env,
+                   lock=True, unlock=False) as cloned_env:
+            cloned_robot = Cloned(robot)
 
-        return traj
+            def call_planner():
+                try:
+                    planner_traj = self.func(instance, cloned_robot,
+                                             *args, **kw_args)
+                    traj = openravepy.RaveCreateTrajectory(
+                        env, planner_traj.GetXMLId()
+                    )
+                    traj.Clone(planner_traj, 0)
+                    return traj
+                finally:
+                    cloned_env.Unlock()
+
+            if defer is True:
```

Comment:
Should this be an explicit comparison against `True`? Or just a check for truthiness.